vm_debug.h

```diff
@@ -94,24 +94,24 @@ bool ruby_debug_log_filter(const char *func_name);
 
 // convenient macro to log even if the USE_RUBY_DEBUG_LOG macro is not specified.
 // You can use this macro for temporary usage (you should not commit it).
-#define _RUBY_DEBUG_LOG(fmt, ...) ruby_debug_log(__FILE__, __LINE__, __func__, fmt, __VA_ARGS__)
+#define _RUBY_DEBUG_LOG(...) ruby_debug_log(__FILE__, __LINE__, __func__, __VA_ARGS__)
 
 #if USE_RUBY_DEBUG_LOG
 
-#define RUBY_DEBUG_LOG(fmt, ...) do { \
+#define RUBY_DEBUG_LOG(...) do { \
   if (ruby_debug_log_mode && ruby_debug_log_filter(__func__)) \
-    ruby_debug_log(__FILE__, __LINE__, __func__, fmt, __VA_ARGS__); \
+    ruby_debug_log(__FILE__, __LINE__, __func__, __VA_ARGS__); \
 } while (0)
```

Comment:
Well this is NG.  With this change

```C
RUBY_DEBUG_LOG()
```

is expanded to

```C
ruby_debug_log(__FILE__, __LINE__, __func__,);
//                                         ^  This
```

The extra comma is ill-formed.

Reply:
See also https://wg21.link/P00306r4

Reply:
Anyway it should be an error, since `ruby_debug_log` requires `fmt` argument?

Reply:
What about this?
```c
#define RUBY_DEBUG_LOG(...) do { \
  if (ruby_debug_log_mode && ruby_debug_log_filter(__func__)) \
    ruby_debug_log(__FILE__, __LINE__, __func__, "" __VA_ARGS__); \
} while (0)
```
`RUBY_DEBUG_LOG()` works as the same as `RUBY_DEBUG_LOG("")`, and it can ensure that `fmt` is a string literal.

Reply:
Ah yes, the `""` hack is clever.